### ISSUE TITLE
finetuning  BUG

### ISSUE THREAD

**Issue by chengchengowen** [BVLC/caffe#2344]
I am finetuning a network with pre-trained model. Parameters are loaded well and the Initialization is seem to be working but failed.

The old versions of code worked well, which used in the same model. While the newest version of code cannot initialize successfully.
 
Could you give me some advice about this problem?

Thanks for your help.

I0421 10:56:48.858861 28571 net.cpp:217] Network initialization done.
I0421 10:56:48.858888 28571 net.cpp:218] Memory required for data: 2334326796
I0421 10:56:48.859872 28571 solver.cpp:42] Solver scaffolding done.
I0421 10:56:48.860260 28571 caffe.cpp:86] Finetuning from ./googlenet_finetune_iter_100000.caffemodel
I0421 10:56:49.588110 28571 solver.cpp:222] Solving GoogleNet
I0421 10:56:49.588150 28571 solver.cpp:223] Learning Rate Policy: poly
I0421 10:56:50.162268 28571 solver.cpp:189] Iteration 0, loss = 108.871
I0421 10:56:50.162325 28571 solver.cpp:204]     Train net output #0: loss1/loss1 = 77.7902 (* 0.3 = 23.3371 loss)
I0421 10:56:50.162343 28571 solver.cpp:204]     Train net output #1: loss2/loss1 = 77.5537 (* 0.3 = 23.2661 loss)
I0421 10:56:50.162369 28571 solver.cpp:204]     Train net output #2: loss3/loss3 = 62.2675 (* 1 = 62.2675 loss)
I0421 10:56:50.162403 28571 solver.cpp:464] Iteration 0, lr = 0.01
I0421 10:57:18.797490 28571 solver.cpp:189] Iteration 40, loss = 13.9469
I0421 10:57:18.797596 28571 solver.cpp:204]     Train net output #0: loss1/loss1 = 8.11429 (* 0.3 = 2.43429 loss)
I0421 10:57:18.797615 28571 solver.cpp:204]     Train net output #1: loss2/loss1 = 8.16322 (* 0.3 = 2.44897 loss)
I0421 10:57:18.797629 28571 solver.cpp:204]     Train net output #2: loss3/loss3 = 8.06469 (* 1 = 8.06469 loss)
I0421 10:57:18.797647 28571 solver.cpp:464] Iteration 40, lr = 0.00999985
I0421 10:57:47.427876 28571 solver.cpp:189] Iteration 80, loss = 12.7271
I0421 10:57:47.427939 28571 solver.cpp:204]     Train net output #0: loss1/loss1 = 8.11129 (* 0.3 = 2.43339 loss)
I0421 10:57:47.427954 28571 solver.cpp:204]     Train net output #1: loss2/loss1 = 8.07988 (* 0.3 = 2.42397 loss)
I0421 10:57:47.427968 28571 solver.cpp:204]     Train net output #2: loss3/loss3 = 7.66401 (* 1 = 7.66401 loss)
I0421 10:57:47.427981 28571 solver.cpp:464] Iteration 80, lr = 0.0099997
I0421 10:58:16.055865 28571 solver.cpp:189] Iteration 120, loss = 12.547
I0421 10:58:16.055975 28571 solver.cpp:204]     Train net output #0: loss1/loss1 = 8.10924 (* 0.3 = 2.43277 loss)
I0421 10:58:16.055994 28571 solver.cpp:204]     Train net output #1: loss2/loss1 = 8.06104 (* 0.3 = 2.41831 loss)
I0421 10:58:16.056006 28571 solver.cpp:204]     Train net output #2: loss3/loss3 = 7.65105 (* 1 = 7.65105 loss)
I0421 10:58:16.056025 28571 solver.cpp:464] Iteration 120, lr = 0.00999956
I0421 10:58:44.670120 28571 solver.cpp:189] Iteration 160, loss = 12.4158
I0421 10:58:44.670174 28571 solver.cpp:204]     Train net output #0: loss1/loss1 = 8.10925 (* 0.3 = 2.43278 loss)
I0421 10:58:44.670192 28571 solver.cpp:204]     Train net output #1: loss2/loss1 = 8.01502 (* 0.3 = 2.40451 loss)
I0421 10:58:44.670205 28571 solver.cpp:204]     Train net output #2: loss3/loss3 = 7.80015 (* 1 = 7.80015 loss)
I0421 10:58:44.670222 28571 solver.cpp:464] Iteration 160, lr = 0.00999941
I0421 10:59:13.284581 28571 solver.cpp:189] Iteration 200, loss = 12.3511
I0421 10:59:13.284771 28571 solver.cpp:204]     Train net output #0: loss1/loss1 = 8.10236 (* 0.3 = 2.43071 loss)
I0421 10:59:13.284790 28571 solver.cpp:204]     Train net output #1: loss2/loss1 = 7.82795 (* 0.3 = 2.34839 loss)
I0421 10:59:13.284808 28571 solver.cpp:204]     Train net output #2: loss3/loss3 = 7.44179 (* 1 = 7.44179 loss)

**Comment by longjon** [comment]
Closing as I can't find a specific issue described here. You're welcome to reopen if you have specific evidence of a problem with Caffe (e.g., you ran command x with commit y and got certain output, but you get different behavior with the same command with `master` in `DEBUG` mode).
